components/DetailPage/ToyInventoryPopUp.js

```diff
@@ -0,0 +1,103 @@
+import styled from "styled-components";
+import StyledButton from "../StyledComponents/StyledButton";
+import { useState } from "react";
+import { toys } from "@/lib/shop";
+import InventoryContainer from "./InventoryContainer";
+import Link from "next/link";
+
+import StyledPopUpOverlay from "@/components/StyledComponents/StyledPopUpOverlay";
+import StyledPopUpContent from "@/components/StyledComponents/StyledPopUpContent";
+import StyledInventoryContainer from "@/components/StyledComponents/StyledInventoryContainer";
+
+const ConfirmPopUpButton = styled(StyledButton)`
+  margin: 0 8px;
+  background-color: ${({ $red }) => $red && "red"};
+  cursor: pointer;
+`;
+
+const ConfirmButtonWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  padding-top: 10px;
+`;
+
+const StyledErrorMessage = styled.p`
+  color: red;
+`;
+
+export default function ToyInventoryPopUp({
+  userStats,
+  onPlayButton,
+  onCancel,
+  petId,
+}) {
+  const [selectedToyItemId, setSelectedToyItemId] = useState(0);
+  const [showErrorMessage, setShowErrorMessage] = useState(false);
+
+  const availableToy = userStats.inventory.toy.filter((toyItems) => {
+    if (toyItems.purchased === true) {
+      return toyItems;
+    }
+  });
+
+  function handleConfirmButtonClick() {
+    if (selectedToyItemId === 0) {
+      setShowErrorMessage(true);
+    } else {
+      onPlayButton(selectedToyItemId);
+      setSelectedToyItemId(true);
+    }
+  }
+
+  function findToyValuesById(id) {
+    const toy = toys.find((toy) => toy.id === id);
+    return toy;
+  }
+
+  function handleClickOnToyItem(id) {
+    setSelectedToyItemId(id);
+    setShowErrorMessage(false);
+  }
+
+  return (
+    <StyledPopUpOverlay>
+      <StyledPopUpContent>
+        <h3>Which toy do you like to use?</h3>
```

Comment:
Slight grammar correction
```suggestion
        <h3>Which toy would you like to use?</h3>
```